source/Lib/vvdec/vvdec.cpp

```diff
@@ -377,16 +377,41 @@ vvdecNalType vvdec_get_nal_unit_type( vvdecAccessUnit *accessUnit )
   return vvdec::VVDecImpl::getNalUnitType(*accessUnit);
 }
 
-const char* vvdec_get_nal_unit_type_name( vvdecNalType t )
+VVDEC_DECL const char* vvdec_get_nal_unit_type_name( vvdecNalType t )
 {
   return vvdec::VVDecImpl::getNalUnitTypeAsString(t);
 }
 
-bool vvdec_is_nal_unit_slice( vvdecNalType t )
+VVDEC_DECL bool vvdec_is_nal_unit_slice( vvdecNalType t )
 {
   return vvdec::VVDecImpl::isNalUnitSlice(t);
 }
 
+VVDEC_DECL int vvdec_set_tracing( const char *file, const char *rule )
+{
+  std::string sTracingFile = file;
+  std::string sTracingRule = rule;
+
+#if ENABLE_TRACING
+  bool bPrint = !( sTracingFile.empty() && sTracingRule.empty() );
+  if( !vvdec::g_trace_ctx )
+    vvdec::g_trace_ctx = vvdec::tracing_init( sTracingFile, sTracingRule );
+
+  if( bPrint && vvdec::g_trace_ctx )
+  {
+    std::string sChannelsList;
+    vvdec::g_trace_ctx->getChannelsList( sChannelsList );
+    vvdec::msg( vvdec::INFO, "\nAvailable tracing channels:\n\n%s\n", sChannelsList.c_str() );
```

Comment:
I would suggest returning `VVDEC_ERR_INITIALIZE`, when printing the channels list. Otherwise it looks good to me.

Reply:
So, don't print the channel list if VVDEC_OK?
(if ENABLE_TRACING is not used, cannot print the list)

Reply:
Oh sorry, I misread your `bPrint` condition. I thought it would only print the channels, when a file but no rule was specified.
Since specifying a file, but no rule doesn't really make sense, I thought we should return an error. But the way it works now is also fine. It just creates an empty trace file.